core/provider/service_test.go

```diff
@@ -846,6 +856,108 @@ func (s *ServiceTestSuite) TestValidateAppeal() {
 
 		actualError := s.service.ValidateAppeal(context.Background(), appeal, provider, policy)
 
+		s.EqualError(actualError, expectedError.Error())
+	})
+
+	s.Run("should return error when required policy question not present", func() {
+		role1 := &domain.Role{ID: "role-1"}
+		appeal := &domain.Appeal{
+			AccountType: "test",
+			Resource: &domain.Resource{
+				ProviderType: mockProviderType,
+			},
+			Role: role1.ID,
+			Options: &domain.AppealOptions{
+				Duration: "24h",
+			},
+		}
+
+		expectedError := fmt.Errorf(`question "%s" is required`, "team")
+
+		provider := &domain.Provider{
+			Config: &domain.ProviderConfig{
+				AllowedAccountTypes: []string{"test"},
+				Appeal: &domain.AppealConfig{
+					AllowPermanentAccess: false,
+				},
+			},
+			Type: mockProviderType,
+		}
+		policy := &domain.Policy{
+			AppealConfig: &domain.PolicyAppealConfig{
+				Questions: []domain.Question{
+					{
+						Key:         "team",
+						Question:    "What team are you in?",
+						Required:    true,
+						Description: "Please provide the name of the team you are in",
+					},
+				},
+			},
+		}
+
+		s.mockProvider.On("GetRoles", mock.Anything, mock.Anything).Return([]*domain.Role{role1}, nil).Once()
+
+		actualError := s.service.ValidateAppeal(context.Background(), appeal, provider, policy)
+
+		s.EqualError(actualError, expectedError.Error())
+	})
+
+	s.Run("should return nil when all valids", func() {
+		role1 := &domain.Role{ID: "role-1"}
+		appeal := &domain.Appeal{
+			AccountType: "test",
+			Resource: &domain.Resource{
+				ProviderType: mockProviderType,
+			},
+			Role: role1.ID,
+			Options: &domain.AppealOptions{
+				Duration: "24h",
+			},
+			Details: map[string]interface{}{
+				provider.ReservedDetailsKeyProviderParameters: map[string]interface{}{
+					"username": "john.doe",
+				},
+				provider.ReservedDetailsKeyPolicyQuestions: map[string]interface{}{
+					"team": "green",
+				},
```

Comment:
Based on discussion with @rahmatrhd, 
We decided to storing answer for 
- [Provider.Config.Parameters](https://github.com/odpf/guardian/blob/173ea0d9a735567c0e2d4b0b506a7804963ca871/domain/provider.go#L76) on `Appeal.Details.__provider_parameters`
- [Policy.AppealConfig.Questions](https://github.com/odpf/guardian/blob/bbcd5eee413e4df77f54be2a7f39ea2f23d8a51a/domain/policy.go#L202) on `Appeal.Details.__policy_questions`

Do you have any concerns or feedbacks @mabdh @ravisuhag @AkarshSatija ?

Reply:
Sounds good.